packages/app/src/background/services/zkIdentity/index.ts

```diff
@@ -189,34 +196,28 @@ export default class ZkIdentityService implements IBackupable {
   private writeConnectedIdentity = async (commitment: string, metadata?: IdentityMetadata): Promise<void> => {
     const ciphertext = this.cryptoService.encrypt(commitment, { mode: ECryptMode.MNEMONIC });
     await this.connectedIdentityStore.set(ciphertext);
+    const connectedMetadata = this.getConnectedIdentityMetadata(metadata);
 
     const [tabs] = await Promise.all([
       browser.tabs.query({ active: true }),
-      pushMessage(
-        setConnectedIdentity({
-          commitment,
-          web2Provider: metadata?.web2Provider,
-          host: metadata?.host,
-        }),
-      ),
+      pushMessage(setConnectedIdentity(connectedMetadata)),
     ]);
 
     await Promise.all(
       tabs.map((tab) =>
-        browser.tabs
-          .sendMessage(
-            tab.id!,
-            setConnectedIdentity({
-              commitment,
-              web2Provider: metadata?.web2Provider,
-              host: metadata?.host,
-            }),
-          )
-          .catch(() => undefined),
+        browser.tabs.sendMessage(tab.id!, setConnectedIdentity(connectedMetadata)).catch(() => undefined),
       ),
     );
   };
 
+  private getConnectedIdentityMetadata(metadata?: IdentityMetadata): ConnectedIdentityMetadata | undefined {
+    if (!metadata) {
```

Comment:
If there is used inside `writeConnectedIdentity` which can clear current connected identity and that's why it can be undefined in `getConnectedIdentityMetadata`.

Reply:
Ok, I will leave a comment in this PR just to get back to this later. #672